### PR TITLE
Implements the editor title in RN for Gutenberg.

### PR DESCRIPTION
**Pull request by diegoreymendez**
## IMPORTANT:

This is a work in progress.  This PR is NOT ready for merging.

## Description:

Implements the editor title in RN for Gutenberg, also removes the native title.

titleinreactnative

## Related PRs:

gutenberg: https://github.com/WordPress/gutenberg/pull/13199
gutenberg-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/450

### Test 1:

1. Run the app.
2. Make sure Gutenberg is enabled.
3. Open a Gutenberg post and edit the title.
4. Update the post, close the editor.

Make sure the post title has been udpated.

### Test 2:

1. Run the app.
2. Make sure Gutenberg is enabled.
3. Open a Gutenberg post and edit the title.
4. Close the editor without saving.

Make sure the post title hasn't changed.

### Test 3:

1. Run the app.
2. Make sure Gutenberg is enabled.
3. Create a new post.

Make sure the placeholder shows up and the title is empty.

### Test 4:

Make sure multiline titles work fine.

### Test 5:

Make sure pressing ENTER does not insert a newline.  Later on we can change this to move the focus to the content but that's outside the scope of this PR.

### CODE REVIEW

**Comment by pinarol** [comment]
When we tap the close button it discards the title changes and closes the editor immediately. 

title2

**Comment by pinarol** [comment]
Another thing that needs to be addressed, we were able to focus on the title field when we create a new post. This behavior also seems to be not working anymore. It's ok if we create a new issue for this to handle later.

title3

File: Podfile [hunk 1]
```diff
@@ -96,7 +96,7 @@ target 'WordPress' do
     ## React Native
     ## =====================
     ##
-    pod 'Gutenberg', :git => 'http://github.com/wordpress-mobile/gutenberg-mobile/', :commit => '704d003adc1f1b76ee0978aa54e05a36fb16371d'
+    pod 'Gutenberg', :git => 'http://github.com/wordpress-mobile/gutenberg-mobile/', :commit => 'e98f6fcf21cac614192ef189e9514ec7f4a03af4'
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply Podfile]: That's correct, but I'd suggest to publish a tag once the review process is complete to avoid multiple tags per PR.

**Comment by pinarol** [comment]
> Another thing that needs to be addressed, we were able to focus on the title field when we create a new post. This behavior also seems to be not working anymore. It's ok if we create a new issue for this to handle later.
> 
@etoledom was this sth you did as a workaround for alpha? (I mean focusing to empty title at start) As far as I remember we need to focus to title at start weather it is empty or not for beta, if so, we can handle all focusing to title issues separately under this issue https://github.com/wordpress-mobile/WordPress-iOS/issues/10594#event-2004410493

File: Podfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (pinarol) [reply Podfile]: was already agreeing on that sorry if I wasn't clear 
> So I think we'll need to update the commit dependencies to a beta tag **before merging our PRs**
Anyway after chatting on this, this conversation might even be outdated

**Comment by pinarol** [comment]
> We followed @iamthomasbishop's design feedback indications and implemented the focusing on keyboard for new posts only.
> 
> [wordpress-mobile/gutenberg-mobile#357 (comment)](https://github.com/wordpress-mobile/gutenberg-mobile/issues/357#issuecomment-448570680)
> 
> I also agree with @diegoreymendez that it would be good to reimplement this behavior on a different PR. 👍

Thanks a lot for clarifying this @etoledom . One small thing, I was quoting myself so I was already agreeing on handling this on a separate PR :)

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 2]
```diff
@@ -280,9 +269,10 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
                                                        callback: callback)
     }
 
-    func gutenbergDidProvideHTML(_ html: String, changed: Bool) {
+    func gutenbergDidProvideHTML(title: String, html: String, changed: Bool) {
         if changed {
             self.html = html
+            self.postTitle = title
```
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: Addressed.

**Comment by diegoreymendez** [comment]
The only remaining issue is the focus-on-open problem.  It's my understanding that this issue will cover it?  https://github.com/wordpress-mobile/WordPress-iOS/issues/10594#event-2004410493

This is ready for another round.

**Comment by iamthomasbishop** [comment]
Looking great @diegoreymendez! Two minor things: 

* I believe the desired title placeholder label is "Add a Title"
* We can probably get rid of the bottom border on the title field, unless we want to leave that for a separate ticket

Also curious how this looks/works in:

* Landscape iPhone
* iPad in landscape

Thanks again, Diego!

**Comment by diegoreymendez** [comment]
@iamthomasbishop - I'll be spawning an issue for the placeholder change.  There's already an issue for the border though.

Those are next in my list of things to-do, but I'll start by merging this PR as-is, to make sure everyone can see the RN title for starters.

**Comment by diegoreymendez** [comment]
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/472

**Pull request merged**